### PR TITLE
Extend HTTP parsing to include path and query

### PR DESCRIPTION
**Pull request by eddyashton**
This adds an `on_url` callback to our `http_parser`, where we extract the path and query. These are forwarded, along with the http method, to the `MsgProcessor`.

I've also moved our "format a HTTP request" util functions into `http_builder.h`, where it should be easier to extend with headers etc if needed,

This is the first stage of #582.

### CODE REVIEW

**Comment by cimetrics** [comment]
images

File: src/enclave/test/http.cpp [hunk 1]
```diff
@@ -3,131 +3,223 @@
...
+    CHECK(m.body == r1);
+  }
+}
+
+TEST_CASE("Method parsing")
+{
+  StubProc sp;
+  enclave::http::Parser p(HTTP_REQUEST, sp);
+
+  bool choice;
```
Comment (jumaffre) [inline src/enclave/test/http.cpp]: Minor: should this be set to `true` or `false` to start with?

Comment (eddyashton) [reply src/enclave/test/http.cpp]: Sure, done!

**Comment by cimetrics** [comment]
images

File: src/enclave/test/http.cpp (hunk 1 above)
Comment (jumaffre) [reply src/enclave/test/http.cpp]: Thanks. It guarantees consistency between test runs, which I think is a good thing.

**Pull request merged**